### PR TITLE
fix: treat header names as case insensitive

### LINKED ISSUE #9: Test framework uses case-sensitive header map implementation
Per https://github.com/GoogleCloudPlatform/service-extensions-samples/pull/8#discussion_r1210721293, our test framework should not use a case-sensitive header map implementation.

### PR DESCRIPTION
**Pull request by martijneken**
Also expand header mutation testing (include updates, removals).

Branched off #38. Fixes #9.

- [x] Tests pass
- [x] Appropriate changes to documentation are included in the PR

### CODE REVIEW

**Comment by leonm1** [comment]
Optional suggestion: since you have commit access to github.com/GoogleCloudPlatform/service-extensions-samples, you can create dependent pull requests that can be reviewed and merged separately.

To do so, you push `martijneken:configured-tests` directly to a branch under GoogleCloudPlatform/service-extensions-samples. Optionally, you can edit #38 to merge that branch into master, but it is not required. Then, you edit this pull request to merge `martijneken:ci-headers` into `martijneken:configured-tests`. That way, I can review this diff separately from #38 while that is not yet merged.

Once #38 is merged, you simply rebase this PR and change the merge target back to main.

If you'd prefer not to do that, I'll wait until #38 is merged to review this.

**Comment by martijneken** [comment]
Yes I would like to learn this skill, but I need some hands on help. I'm having trouble just rebasing this branch onto the latest commit in configured-tests. Let's chat tomorrow.

**Comment by martijneken** [comment]
Rebased! PTAL.

File: plugins/samples/add_header/tests.textpb
Comment (leonm1) [inline plugins/samples/add_header/tests.textpb:65]: nit: even after reading the code, this comment is confusing to me:

At first, I thought this comment was referring to the fact that the "assertion framework" / "config driven testing framework" ignored the case, but I now see that this is referring to the wasm host interface is what is ignoring the case.

But the wasm plugin itself can't ignore case, since if the wasm plugin enumerates all the headers, it will see case-sensitive strings, right?

But the test framework ignores the case set by the plugin in just the assertions and in get and set operations done by the plugin.

Could you take a whack at restating this comment?

**Review by leonm1 — APPROVED** [review]

File: plugins/samples/add_header/tests.textpb
Comment (martijneken) [reply plugins/samples/add_header/tests.textpb]: Yup, good call. Removed this test and added a clearer comment.

**Pull request merged**
